honeybee_energy/programtype.py

```diff
@@ -63,5 +82,567 @@ def name(self):
+    def gas_equipment(self, value):
+        if value is not None:
+            assert isinstance(value, GasEquipment), 'Expected GasEquipment ' \
+                'object for ProgramType.gas_equipment. Got {}.'.format(type(value))
+        self._gas_equipment = value
+
+    @property
+    def infiltration(self):
+        """Get or set an Infiltration object to describe the outdoor air leakage."""
+        return self._infiltration
+
+    @infiltration.setter
+    def infiltration(self, value):
+        if value is not None:
+            assert isinstance(value, Infiltration), 'Expected Infiltration ' \
+                'object for ProgramType.infiltration. Got {}.'.format(type(value))
+        self._infiltration = value
+
+    @property
+    def ventilation(self):
+        """Get or set a Ventilation object to describe the minimum outdoor air flow."""
+        return self._ventilation
+
+    @ventilation.setter
+    def ventilation(self, value):
+        if value is not None:
+            assert isinstance(value, Ventilation), 'Expected Ventilation ' \
+                'object for ProgramType.ventilation. Got {}.'.format(type(value))
+        self._ventilation = value
+
+    @property
+    def setpoint(self):
+        """Get or set a Setpoint object to describe the temperature setpoints."""
+        return self._setpoint
+
+    @setpoint.setter
+    def setpoint(self, value):
+        if value is not None:
+            assert isinstance(value, Setpoint), 'Expected Setpoint ' \
+                'object for ProgramType.setpoint. Got {}.'.format(type(value))
+        self._setpoint = value
+
+    @property
+    def schedules(self):
+        """List of all schedules contained within the ProgramType."""
+        sched = []
+        if self.people is not None:
+            sched.append(self.people.occupancy_schedule)
+            sched.append(self.people.activity_schedule)
+        if self.lighting is not None:
+            sched.append(self.lighting.schedule)
+        if self.electric_equipment is not None:
+            sched.append(self.electric_equipment.schedule)
+        if self.gas_equipment is not None:
+            sched.append(self.gas_equipment.schedule)
+        if self.infiltration is not None:
+            sched.append(self.infiltration.schedule)
+        if self.ventilation is not None and self.ventilation.schedule is not None:
+            sched.append(self.ventilation.schedule)
+        if self.setpoint is not None:
+            sched.append(self.setpoint.heating_schedule)
+            sched.append(self.setpoint.cooling_schedule)
+            if self.setpoint.humidifying_schedule is not None:
+                sched.append(self.setpoint.humidifying_schedule)
+                sched.append(self.setpoint.dehumidifying_schedule)
+        return sched
+
+    @property
+    def unique_schedules(self):
+        """List of all unique schedules contained within the ProgramType."""
+        return list(set(self.schedules))
```

Comment:
I would say the `schedules` property should return the unique list of schedules. Do you see a use case to get the list with duplicate schedules?

Reply:
There were two reasons why I have separate properties for both schedules and unique schedule:
1. It parallels the `ConstructionSet`, which has properties for both `constructions`, which returns a list of all constructions where you know what type of construction it is by its index, and `unique_constructions`, which only gives you the unique ones and where the index doesn't tell you anything (but this is needed for the Model).
2, You are right that it's rare to get the same types of schedules within a single ProgramType (it's far more common to have shared schedules between program types). For this reason, I don't usually want to go through the whole process of hashing the schedule objects to create a python set if I don't need to (this hashing can definitely add up over large models).  However, there are definitely cases where the same schedule is used for two fields (like occupancy and ventilation) and so I need a `unique_schedules` object to make sure that I only pull the unique ones for these cases.

Reply:
I can't really follow your logic here but I trust your judgment on the topic. Let's keep both.